plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/functions/tablechanges/TableChangesFunction.java

```diff
@@ -0,0 +1,172 @@
+import static java.util.Objects.requireNonNull;
+
+public class TableChangesFunction
+        extends AbstractConnectorTableFunction
+{
+    private static final String SCHEMA_NAME = "system";
+    private static final String NAME = "table_changes";
+    public static final SchemaFunctionName TABLE_CHANGES_NAME = new SchemaFunctionName(SCHEMA_NAME, NAME);
+    public static final String TABLE_SCHEMA_NAME_ARGUMENT = "TABLE_SCHEMA_NAME";
+    private static final String TABLE_NAME_ARGUMENT = "TABLE_NAME";
+    private static final String SINCE_VERSION_ARGUMENT = "SINCE_VERSION";
+    private static final String CHANGE_TYPE_COLUMN_NAME = "_change_type";
+    private static final String COMMIT_VERSION_COLUMN_NAME = "_commit_version";
+    private static final String COMMIT_TIMESTAMP_COLUMN_NAME = "_commit_timestamp";
+    private static final Set<String> IGNORED_COLUMNS = ImmutableSet.of(PATH_COLUMN_NAME, FILE_SIZE_COLUMN_NAME, FILE_MODIFIED_TIME_COLUMN_NAME);
+
+    private final TrinoFileSystemFactory fileSystemFactory;
+    private final DeltaLakeMetadataFactory deltaLakeMetadataFactory;
+
+    public TableChangesFunction(TrinoFileSystemFactory fileSystemFactory, DeltaLakeMetadataFactory deltaLakeMetadataFactory)
+    {
+        super(
+                SCHEMA_NAME,
+                NAME,
+                ImmutableList.of(
+                        ScalarArgumentSpecification.builder().name(TABLE_SCHEMA_NAME_ARGUMENT).type(VARCHAR).build(),
+                        ScalarArgumentSpecification.builder().name(TABLE_NAME_ARGUMENT).type(VARCHAR).build(),
+                        ScalarArgumentSpecification.builder().name(SINCE_VERSION_ARGUMENT).type(BIGINT).defaultValue(-1L).build()),
+                GENERIC_TABLE);
+        this.fileSystemFactory = requireNonNull(fileSystemFactory, "fileSystemFactory is null");
+        this.deltaLakeMetadataFactory = requireNonNull(deltaLakeMetadataFactory, "deltaLakeMetadataFactory is null");
+    }
+
+    @Override
+    public TableFunctionAnalysis analyze(ConnectorSession session, ConnectorTransactionHandle transaction, Map<String, Argument> arguments)
+    {
+        ScalarArgument schemaNameArgument = (ScalarArgument) arguments.get(TABLE_SCHEMA_NAME_ARGUMENT);
+        checkFunctionArgument(schemaNameArgument.getValue() != null, "schema_name value for function table_changes() cannot be null");
+        String schemaName = ((Slice) schemaNameArgument.getValue()).toStringUtf8();
+
+        ScalarArgument tableNameArgument = (ScalarArgument) arguments.get(TABLE_NAME_ARGUMENT);
+        checkFunctionArgument(tableNameArgument.getValue() != null, "table_name value for function table_changes() cannot be null");
+        String tableName = ((Slice) tableNameArgument.getValue()).toStringUtf8();
+
+        ScalarArgument sinceVersionArgument = (ScalarArgument) arguments.get(SINCE_VERSION_ARGUMENT);
+        long sinceVersion = (Long) sinceVersionArgument.getValue() + 1; //+1 to ensure that the sinceVersion is exclusive
```

Comment:
Let's check that the argument is not `NULL` before getting its value.
Could you please add a set of failure tests to cover the combinations of calling the `table_changes` function with `NULL` / missing parameters?

Reply:
Missing parameters is covered in few places. I will add a line to `testReadingRangesOfChangeOnTableWithCdfEnabled` that checks for null case